derive_client/utils/w3.py

```diff
@@ -108,17 +109,15 @@ def send_and_confirm_tx(
     *,
     action: str,  # e.g. "approve()", "deposit()", "withdraw()"
 ) -> TxResult:
-    """Send and confirm transactions, after simulating."""
-
-    tx_result = TxResult(tx_hash="", tx_receipt=None, exception=None)
+    """Send and confirm transactions."""
 
     try:
         tx_hash = sign_and_send_tx(w3=w3, tx=tx, private_key=private_key)
-        tx_result.tx_hash = tx_hash.hex()
+        tx_result = TxResult(tx_hash=tx_hash.to_0x_hex(), tx_receipt=None, exception=None)
     except Exception as send_err:
-        print(f"❌ Failed to send tx for {action}, error: {send_err!r}")
-        tx_result.exception = send_err
-        return tx_result
+        msg = f"❌ Failed to send tx for {action}, error: {send_err!r}"
+        print(msg)
+        raise TxSubmissionError(msg) from send_err
```

Comment:
we raise, since there won't be a `tx_hash` so there cannot be a `TxResult`. Propagating this onto `BridgeTxResult` was not good either imo. If we fail to even create the transaction, this will be an exception beyond the scope of the  `BridgeTxResult`, so rather raise